File: AlgorithmsProject/src/in/designpatterns/scala/factory/PizzaIngrediantFactory.scala
package in.designpatterns.scala.factory
import in.designpatterns.scala.factory.ingredients._
trait PizzaIngrediantFactory {
	def createCheese() : Cheese
	def createClam() : Clam
	def createDought() : Dough
	def createPepperoni() : Pepperoni
	def createSauce() : Sauce
	def createVeggies : Set[Veggies]
}

File: AlgorithmsProject/src/in/designpatterns/scala/factory/PizzaIngrediantFactory.java
package in.designpatterns.scala.factory;

import in.designpatterns.scala.factory.ingredients.*;

public interface PizzaIngrediantFactory {
    Dough createDough();
    Sauce createSauce();
    Cheese createCheese();
    Veggies[] createVeggies();
    Pepperoni createPepperoni();
    Clam createClams();
}
